source/code/plugins/VMInsightsIDataCollector.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+module VMInsights
+
+    # Design assumptions:
+    # 1. Changing the number of active CPUs requires a system reboot.
```

Comment:
On windows we are enumerating the number of cpus and memory size every minute.

Reply:
To the best of my knowledge, neither the number of active CPUs nor the total memory can change without a reboot. If we can find evidence to contradict that, we'll need to also change how the calculations are done.